src/api/app/mixins/get_flags.rb

```diff
@@ -0,0 +1,34 @@
+module GetFlags
+  # FIXME: All of this would not be needed if creating an Package/Project would create all
+  #        the build, publish, debuginfo and useforbuild Flags. The only reason
+  #        this happens here is because rendering the XML does ignore the defaults.
+  # Returns a hash of arrays, sorted by repository.
+  # The arrays contain Flag objects of type, sorted by architecture.
+  # Like:
+  # {all: [Flag, Flag-i586, Flag-x86_64], 13.2: [Flag, Flag-i585, Flag-x86_64]}
+  def get_flags(flag_type)
+    the_flags = {}
+
+    # FIXME: A helper called in a model...
+    return the_flags unless FlagHelper::TYPES.has_key?(flag_type.to_s)
+
+    # [nil] is for no architecture/repository
+    ['all'].concat(self.repositories.map{|repo| repo.name}).each do |repository|
+      the_flags[repository] = []
+      ['all'].concat(self.architectures.uniq).each do |architecture|
+        arch = ( architecture == 'all' ) ? nil : architecture.id
+        architecture = nil if architecture == 'all'
+        repo = ( repository == 'all' ) ? nil : repository
+        flag = self.flags.where(flag: flag_type).where(repo: repo).where(architecture_id: arch).first
```

Comment:
How about this?

```ruby
# Nil is a placeholder for "all" architectures
[nil].concat(self.architectures.uniq).each do |architecture|
flag = self.flags.where(flag: flag_type).where(repo: repo).where(architecture: architecture).first
```

I don't like the [nil] thing either. But it seems easier to read to me than swapping around nil and "all" like before. And "all" isn't actually used.